src/components/sbb-menu/sbb-menu.tsx

```diff
@@ -0,0 +1,292 @@
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'sbb-menu_did-open',
+  })
+  public didOpen: EventEmitter<void>;
+
+  /**
+   * Emits whenever the menu begins the closing transition.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'sbb-menu_will-close',
+  })
+  public willClose: EventEmitter<void>;
+
+  /**
+   * Emits whenever the menu is closed.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'sbb-menu_did-close',
+  })
+  public didClose: EventEmitter<void>;
+
+  /**
+   * Opens the menu on trigger click.
+   */
+  @Method()
+  public async open(): Promise<void> {
+    this.willOpen.emit();
+    this._setMenuPosition();
+    this._dialog.showModal();
+  }
+
+  /**
+   * Closes the menu.
+   */
+  @Method()
+  @Listen('sbb-menu-action_click')
+  public async close(): Promise<void> {
+    this.willClose.emit();
+    this._isDismissing = true;
+  }
+
+  // Resets position on window resize.
+  @Listen('resize', { target: 'window', passive: true })
+  public onResizeWindowEvent(): void {
+    if (this._open) {
+      this._setMenuPosition();
+    }
+  }
+
+  // Closes the menu on "Esc" key pressed.
+  @Listen('keydown', { target: 'window' })
+  public onEscAction(event: KeyboardEvent): void {
+    if (this._open && event.key === 'Escape') {
+      event.preventDefault();
+      this.close();
+    }
+  }
+
+  // Removes trigger click listener on trigger change.
+  @Watch('trigger')
+  public removeTriggerClickListener(
+    newValue: string | HTMLElement,
+    oldValue: string | HTMLElement
+  ): void {
+    if (newValue !== oldValue) {
+      this._triggerController?.abort();
+      this._configureTriggerElement(this.trigger);
+    }
+  }
+
+  // Internal dialog element.
+  private _dialog: HTMLDialogElement;
+
+  // The element that triggers the open of the sbb-menu.
+  private _triggerEl: HTMLElement;
+
+  // Controller used to abort the open of the abb-menu on trigger change.
+  private _triggerController: AbortController;
+
+  public componentDidLoad(): void {
+    this._dialog = this.el.shadowRoot.querySelector('dialog');
+
+    // validate trigger element and attach click event
+    this._configureTriggerElement(this.trigger);
+
+    // close menu on interactive elements click
+    this._dismissOnInteractiveElementClick();
+
+    // close menu on backdrop click
+    this._dismissOnBackdropClick();
+
+    // listen for menu animation to end
+    this._dialog.onanimationend = (event: AnimationEvent) => this._onMenuAnimationEnd(event);
+  }
+
+  // Check if the trigger is valid and attach click event listener.
+  private _configureTriggerElement(trigger: string | HTMLElement): void {
+    if (!trigger) {
+      return;
+    }
+
+    // check whether it's a string or an HTMLElement
+    if (typeof trigger === 'string') {
+      this._triggerEl = document.getElementById(trigger);
+    } else if (trigger instanceof window.Element) {
```

Comment:
Is the check mandatory? Since `trigger: string | HTMLElement`, if it's not a string, it must be the latter.

Reply:
Yes, as we have already discussed it could also be passed an invalid value or object so maybe it is worth leaving the control in any case.